src/test/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogitSpec.scala

```diff
@@ -0,0 +1,121 @@
+package beam.agentsim.agents.choice.mode
+
+import scala.util.Random
+
+import beam.agentsim.agents.choice.mode.ModeChoiceMultinomialLogitSpec.{randomIntList, randomPath}
+import beam.agentsim.events.SpaceTime
+import beam.router.model.BeamPath
+import beam.router.model.RoutingModel.TransitStopsInfo
+import beam.router.r5.{BikeLanesAdjustment, BikeLanesData}
+import org.matsim.api.core.v01.Id
+import org.matsim.vehicles.Vehicle
+import org.scalatest.WordSpec
+
+class ModeChoiceMultinomialLogitSpec extends WordSpec {
+
+  "ModeChoiceMultinomialLogit" when {
+    "pathScaledForWaiting" must {
+      "skip the first element of BeamPath and sum all travelTime when bikeLaneScaleFactor is 1" in {
+        val value1 = 11.4D
+        val value2 = 12.3D
+        val path = randomPath(travelTimes = 10D, value1, value2)
+        val data = BikeLanesData(scaleFactorFromConfig = 1D, bikeLanesLinkIds = randomIntList.toSet)
+        val adjustment = new BikeLanesAdjustment(data)
+
+        val result: Double = ModeChoiceMultinomialLogit.pathScaledForWaiting(path, adjustment)
+
+        assert(result === value1 + value2)
+      }
+
+      "be zero when path has zero or only one element" in {
+        val path = if (Random.nextBoolean()) randomPath() else randomPath(travelTimes = 10D)
+        val data = BikeLanesData(scaleFactorFromConfig = Random.nextDouble(), bikeLanesLinkIds = randomIntList.toSet)
+        val adjustment = new BikeLanesAdjustment(data)
+
+        val result: Double = ModeChoiceMultinomialLogit.pathScaledForWaiting(path, adjustment)
+
+        assert(result === 0)
+      }
+
+      "pathScaledForWaiting scale disproportionately linkIds that are specified on BikeLaneAdjustment" in {
+        val value1 = 11.4D
+        val value2 = 12.3D
+        val sharedLinkId1 = 2
+        val nonSharedLinkId = 3
+        val linkIds = Seq(1, sharedLinkId1, nonSharedLinkId)
+        val path = randomPath(travelTimes = Seq(10D, value1, value2), linkIds)
+        val data = new BikeLanesData(
+          scaleFactorFromConfig = 2D,
+          bikeLanesLinkIds = Set(sharedLinkId1, nonSharedLinkId + 1)
+        )
+        val adjustment = new BikeLanesAdjustment(data)
+
+        val result: Double = ModeChoiceMultinomialLogit.pathScaledForWaiting(path, adjustment)
+
+        assert(result === value1 * 1 / data.scaleFactorFromConfig + value2)
+      }
+
+    }
+  }
+
+}
+
+object ModeChoiceMultinomialLogitSpec {
```

Comment:
nitpick: no seed number for the test which means that every time I run it the input will change. Isn't it against the general TDD concept? I'm fine to use them in Fuzzy testing, but it is not the case

Reply:
TDD is the process (is not against randomization). Introducing limited randomization help to figure corner cases and increase the readability since makes clear what we are testing. Random data means "I don't care about those values" or "can be any value that belongs to this domain"

Reply:
Without providing seed number (which is the only thing which will help you to find a corner cases) your test is not reproducible - if it fails I have no idea what was an input which causes it to fail. And I don't see currently any extra code to say what was the whole input.

Regarding your input, once in a while it will fail because it will generate random number which can be Int.MaxValue, and the sum of two Int.MaxValue is -2 in Scala.

Reply:
> Regarding your provided example with anyPositiveNumber, once in a while it will fail because it will generate a random number which can be Int.MaxValue, and the sum of two Int.MaxValue is -2 in Scala.

![image](https://user-images.githubusercontent.com/12264098/91591598-890a9f00-e933-11ea-8b4e-adf7d3115663.png)

This is the benefit I talked about

Reply:
@carloscaldas but how will I figure out if something is wrong, I'll just get an assertion and have no idea of the whole input. TDD is traditionally used with deterministic algorithms when there is a known input and one expected result. Good unit tests should be reproducible and independent from external factors such as the environment (current time, random number generator and so on) or running order. Introducing randomness doesn't help me to understand whether the input is wrong or the code. Again, if it is fuzzy testing, I'm fine, but I don't think the current implementation is fuzzy testing.

Reply:
So many thoughts:

* With regards to reproducibility - I agree that this should be reproducible. And the way to do that is by outputting the random values so that it **can** then be reproduced. Often you only need to output those random values on failure, though. 
* With regards to randomness - that should not be part of the tests that are run every commit. The randomness can make flaky tests and shakiness with the tests overall. It can be debated whether they should be part of the final PR review is a good question, though. The two options would be to have it as a daily standalone test to make sure develop hasn't regressed and/or find new bugs -- or to have it as a second requirement to a PR which would only need run when you feel the PR is ready for final review.
* If this is a one off, then that's fine, but we could also look into using Scalacheck - which has built in utilities for really good randomness - https://medium.com/se-notes-by-alexey-novakov/generate-random-data-with-scalacheck-a89ba699fe3d (and it outputs the random factors by default iirc)

Ultimately, bullet number 1 needs done for this code imho.